app.rb

```diff
@@ -0,0 +1,158 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+
+class App
+  def run
+    puts 'Welcome to School Library App!'
+    loop do
+      puts '-' * 50
+      puts '
+          1- List all books
+          2- List all people
+          3- Create a new person
+          4- Create a new book entry
+          5- Create a new rental entry
+          6- List all rentals for a given person id
+          7- Quit'
+      puts 'Choose an option: '
+      input = gets.chomp.to_i
+
+      break if input == 7
+
+      operation(input)
+    end
+  end
+
+  def operation(input)
+    case input
+    when 1
+      list_books
+    when 2
+      list_people
+    when 3
+      create_person
+    when 4
+      create_book
+    when 5
+      create_rental
+    when 6
+      list_rentals
+    else
+      puts 'Please choose a valid option number'
+    end
+  end
+
+  def list_books
+    puts '-' * 50
+    books = Book.class_variable_get(:@@books)
+    if books.empty?
+      puts 'The books list is empty'
+    else
+      puts 'Books list:'
+      books.each_with_index do |book, index|
+        puts "[Book #{index + 1}] Title: #{book.title} | Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_people
+    puts '-' * 50
+    people = Person.class_variable_get(:@@people)
+    if people.empty?
+      puts 'The list is empty'
+    else
+      puts 'People list:'
+      people.each_with_index do |person, index|
+        puts
+        "[Teacher #{index + 1}] ID: #{person.id} | Name: #{person.name} | Age: #{person.age}" if person.is_a?(Teacher)
+        puts
+        "[Student #{index + 1}] ID: #{person.id} | Name: #{person.name} | Age: #{person.age}" if person.is_a?(Student)
+      end
+    end
+  end
```

Comment:
- [ ] There is a problem with the `list_people` method as it doesn't display correctly. Please see error below and fix it 👇 
![image](https://user-images.githubusercontent.com/46839548/180455508-79ed37f8-96de-4f82-bf41-4671d33c8cb5.png)

- [ ] As a result of the error mentioned above ☝️, it is impossible to create a new rental entry because the people list is not displaying. Kindly see the below figure 👇 
![image](https://user-images.githubusercontent.com/46839548/180456420-1f484415-dea0-4da6-94f2-a67e4f3bf1b1.png)

- [ ] Since we can't see the people list, and we can't create a new rental, it is impossible to list all rentals for a given person's ID. So, let's try to fix the `list_people` method since most other things depends on it displaying correctly 😎